libs/gpg45-evaluator/src/test/java/uk/gov/di/ipv/core/library/gpg45/domain/CredentialEvidenceItemTest.java

```diff
@@ -0,0 +1,402 @@
+        return Stream.of(
+                CredentialEvidenceItem.builder().build(),
+                CredentialEvidenceItem.builder().activityHistoryScore(10).strengthScore(10).build(),
+                CredentialEvidenceItem.builder()
+                        .identityFraudScore(10)
+                        .verificationScore(10)
+                        .build());
+    }
+
+    @Test
+    void shouldGetTypeNino() throws UnknownEvidenceTypeException {
+        CredentialEvidenceItem credentialEvidenceItem1 =
+                CredentialEvidenceItem.builder()
+                        .checkDetails(List.of(new CheckDetail()))
+                        .failedCheckDetails(List.of(new CheckDetail()))
+                        .build();
+        assertEquals(EvidenceType.NINO, credentialEvidenceItem1.getType());
+
+        CredentialEvidenceItem credentialEvidenceItem2 =
+                CredentialEvidenceItem.builder()
+                        .failedCheckDetails(List.of(new CheckDetail()))
+                        .build();
+        assertEquals(EvidenceType.NINO, credentialEvidenceItem2.getType());
+
+        CredentialEvidenceItem credentialEvidenceItem3 =
+                CredentialEvidenceItem.builder().checkDetails(List.of(new CheckDetail())).build();
+        assertEquals(EvidenceType.NINO, credentialEvidenceItem3.getType());
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideCasesForNotNino")
+    void shouldNotGetTypeNino(CredentialEvidenceItem item) {
+        try {
+            assertNotEquals(EvidenceType.NINO, item.getType());
+        } catch (UnknownEvidenceTypeException exception) {
+            assertThrows(UnknownEvidenceTypeException.class, item::getType);
+        }
+    }
+
+    private static Stream<CredentialEvidenceItem> provideCasesForNotNino() {
+        return Stream.of(
+                CredentialEvidenceItem.builder().strengthScore(10).build(),
+                CredentialEvidenceItem.builder().validityScore(10).build(),
+                CredentialEvidenceItem.builder().verificationScore(10).build(),
+                CredentialEvidenceItem.builder().activityHistoryScore(10).build(),
+                CredentialEvidenceItem.builder().identityFraudScore(10).build());
+    }
+
+    @Test
+    void throwsForGetTypeUnknown() {
+        CredentialEvidenceItem credentialEvidenceItem = CredentialEvidenceItem.builder().build();
+        assertThrows(UnknownEvidenceTypeException.class, credentialEvidenceItem::getType);
+    }
+
+    @Test
+    void shouldGetEvidenceScore() {
+        CredentialEvidenceItem credentialEvidenceItem =
+                CredentialEvidenceItem.builder().strengthScore(10).validityScore(20).build();
+        assertEquals(new Gpg45Scores.Evidence(10, 20), credentialEvidenceItem.getEvidenceScore());
+    }
+
+    @Test
+    void shouldHasContraIndicators() {
+        CredentialEvidenceItem credentialEvidenceItem1 = new CredentialEvidenceItem(10, 20, null);
```

Comment:
I think, if we are not writing constructor test, I think using a builder() would be a great choice.